api/v1/cluster_types.go

```diff
@@ -2333,20 +2348,37 @@ type ExternalCluster struct {
 	BarmanObjectStore *BarmanObjectStoreConfiguration `json:"barmanObjectStore,omitempty"`
 }
 
-// AppendAdditionalCommandArgs adds custom arguments as barman cloud command-line options
-func (cfg *BarmanObjectStoreConfiguration) AppendAdditionalCommandArgs(options []string) []string {
-	if cfg == nil || cfg.Data == nil {
+// AppendAdditionalCommandArgs adds custom arguments as barman-cloud-backup command-line options
+func (cfg *DataBackupConfiguration) AppendAdditionalCommandArgs(options []string) []string {
+	if cfg == nil {
+		return options
+	}
+	return appendAdditionalCommandArgs(cfg.AdditionalCommandArgs, options)
+}
+
+// AppendAdditionalCommandArgs adds custom arguments as barman-cloud-wal-archive command-line options
+func (cfg *WalBackupConfiguration) AppendAdditionalCommandArgs(options []string) []string {
+	if cfg == nil {
 		return options
 	}
+	return appendAdditionalCommandArgs(cfg.AdditionalCommandArgs, options)
+}
 
-	for _, userOption := range cfg.Data.AdditionalCommandArgs {
+func appendAdditionalCommandArgs(additionalCommandArgs []string, options []string) []string {
+	optionsKeys := map[string]bool{}
```

Comment:
If you don't mind, I would like rename `optionsKeys` --> `optionKeys` and `userOption` --> `additionalCommandArg`.